### ISSUE TITLE
Outdated dependencies

### ISSUE THREAD

**Issue by bjgill** [Metaswitch/swagger-rs#51]
We currently have a few outdated dependencies:

* hyper - will require some major changes to update to v0.12 from v0.11
* uuid - trivial
* base64 - trivial

I've not yet worked out whether any of these would not be breaking changes. Thus, my vague plan is to wait until we get round to updating hyper, and sweep up all the other upgrades at the same time.

**Comment by CodeSandwich** [comment]
I'd suggest upgrading yet another dependency, which is burried deep in other outdated crates: openssl. Current version (0.9.24) does not compile on systems with OpenSSL 1.1.1.

**Comment by richardwhiuk** [comment]
Sadly we can't update OpenSSL to a later version.

In native-tls 0.2, which supports Rust OpenSSL 0.10, they've dropped support for `TlsConnectorBuilderExt::from_openssl`, which we use in https://github.com/Metaswitch/swagger-rs/blob/f86922a3b9a6967831faa891922f0bedf38d0fec/src/connector.rs#L88-L99

Until we add sufficient function to native TLS to do what we are doing by using OpenSSL manually, we can't move.

**Comment by richardwhiuk** [comment]
I've raised https://github.com/sfackler/rust-native-tls/issues/125 to look at getting a fix to unblock this.

**Comment by richardwhiuk** [comment]
I'm going to close this, and raise a separate issue for the OpenSSL and hyper-tls issues.

The ones listed in @bjgill's original post are resolved in 3.0
